### PR TITLE
Fix Expense Duplication and Screen Flicker in FinanceViewModel and CreateExpense Screen

### PR DESCRIPTION
**Pull request by tosigame**
This PR addresses two critical issues in the FinanceViewModel and CreateExpense screen. Firstly, it resolves the expense duplication bug by ensuring addExpense is not called multiple times when invoked rapidly. Secondly, it fixes the screen flicker issue by properly loading data before displaying the expense list, preventing the initial empty state. Unit tests have been added to verify the bug.

### CODE REVIEW

File: app/src/test/java/com/github/se/wanderpals/viewmodel/FinanceViewModelTest.kt
Comment (mtv47) [inline app/src/test/java/com/github/se/wanderpals/viewmodel/FinanceViewModelTest.kt:1]: The test is very thought out and proves that the fix is really working. Great job in putting in the effort.

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/FinanceViewModel.kt [hunk 1]
```diff
@@ -83,14 +91,31 @@ open class FinanceViewModel(val tripsRepository: TripsRepository, val tripId: St
...
+        if (expenseList.isEmpty()) {
+          // error
+          Log.e("FinanceViewModel", "Error reading expense after addition (should not happen)")
+        } else {
+          val newExpense = expenseList.last()
+          NotificationsManager.addExpenseNotification(tripId, newExpense)
+        }
+        updateStateLists()
+        _isAddingExpense.value = false
+        _createExpenseFinished.value = true
```
Comment (mtv47) [inline app/src/main/java/com/github/se/wanderpals/model/viewmodel/FinanceViewModel.kt:109]: Could have used setCreateExpenseFinished for more readability

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/FinanceViewModel.kt [hunk 2]
```diff
@@ -83,14 +91,31 @@ open class FinanceViewModel(val tripsRepository: TripsRepository, val tripId: St
...
+
     viewModelScope.launch {
-      val newExpense = tripsRepository.getAllExpensesFromTrip(tripId).last()
-      NotificationsManager.addExpenseNotification(tripId, newExpense)
-      updateStateLists()
+      if (!isAddingExpense.value && !createExpenseFinished.value) {
+        _isAddingExpense.value = true
+        tripsRepository.addExpenseToTrip(tripId, expense)
+        val expenseList = tripsRepository.getAllExpensesFromTrip(tripId)
+        if (expenseList.isEmpty()) {
```
Comment (mtv47) [inline app/src/main/java/com/github/se/wanderpals/model/viewmodel/FinanceViewModel.kt:100]: Why not, more error checking is always good. Isn't relevant to this PR but is appreciated

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/finance/CreateExpense.kt
Comment (mtv47) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/trip/finance/CreateExpense.kt:1]: Clean

**Review by mtv47 — APPROVED** [review]
Efficient and nice PR. Fixes a bug that is important with additional written tests that check that the bug won't appear further down the line

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/FinanceViewModel.kt (hunk 1 above)
Comment (tosigame) [reply app/src/main/java/com/github/se/wanderpals/model/viewmodel/FinanceViewModel.kt:109]: You may be right, however my idea behind setCreateExpenseFinished, was for the UI to signal to the viewModel, that it has finished getting the information from the update.

**Pull request merged**